### DISCUSSION TITLE
v1.79.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#4164] [Announcements]
# v1.79.0

## Highlights

Welcome to release `v1.79.0` of Immich. This release introduces the most requested feature since the beginning of Immich; the ability to **serve existing photos** in a streamlined and easy-to-use manner. Not only that, this release packs a ton of other improvements that we hope you will like. Some of the highlights below

- Support for external libraries
- Set minimum detected face to show on your instance
- Notable fix: Video EXIF parsing
- Notable fix: Android background upload regression
- Notable fix: Jerky mobile's swipe behavior

### Existing library support

Thanks to many months of hard work from @etnoy, @jrasm91, @danieldietzler. In this release, all assets will be part of a library. Existing assets are put in an “upload library” where all mobile uploads will be put. You can create an external library if you want to point Immich to your existing photos. Some basic features of the library mechanism are

- Supports creating multiple external libraries.
- Each library can recursively import one or more paths. 
- Add exclude paths from being imported.

Complete documentation on how to use external libraries can be viewed [here](https://immich.app/docs/features/libraries)

Libraries, being a new top-level container, also allow us to develop more exciting features in the future.

> [!NOTE]  
> Support for read-only [imported assets](https://immich.app/docs/features/read-only-gallery) has not changed with this release, although it will likely be deprecated soon. 

<p align="center">
<img width="808" alt="Library List" src="https://github.com/immich-app/immich/assets/27055614/3cc5d4f8-8f7a-47e0-912a-582fcf4610f8">
</p>

> [!NOTE]  
> Please help us test and report any bugs that you find in this new feature. We are looking forward to receiving your feedback and improving in the subsequent releases.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Server
* feat(server,web): libraries by @etnoy in https://github.com/immich-app/immich/pull/3124
* fix(server): use libopus for transcoding by @mertalev in https://github.com/immich-app/immich/pull/4102
* fix(server): use ExifTool decoded values and unify metadata extraction by @uhthomas in https://github.com/immich-app/immich/pull/2908
* fix(server): Error when loading album with deleted owner by @danieldietzler in https://github.com/immich-app/immich/pull/4086
* feat(server): add m4v format by @jrasm91 in https://github.com/immich-app/immich/pull/4135
* chore(server): Use access core for person permissions by @danieldietzler in https://github.com/immich-app/immich/pull/4138
* feat(server): add setting for minimum face count for face detection by @GenericGuy in https://github.com/immich-app/immich/pull/4128

### Web
* feat(web): hide face from detail page by @martabal in https://github.com/immich-app/immich/pull/4098
* fix(web): Context menu underflowing by @danieldietzler in https://github.com/immich-app/immich/pull/4127
* fix(web): Assigns description text area with asset description if it exists #4073 by @sellnat77 in https://github.com/immich-app/immich/pull/4125
* feat(web): swap between people when merging faces by @martabal in https://github.com/immich-app/immich/pull/4089
* feat(web): cancel pending requests on image change by @lmichelin in https://github.com/immich-app/immich/pull/4145
* fix(web): revert assigns description text area with asset description if it exists by @martabal in https://github.com/immich-app/immich/pull/4160
* fix(web): clickable accordion layout by @Ploonet in https://github.com/immich-app/immich/pull/4129
* feat(web): make settings accordion click area larger by @Ploonet in https://github.com/immich-app/immich/pull/4119

### Mobile
* feat(mobile): image caching & viewer improvements by @fyfrey in https://github.com/immich-app/immich/pull/4095
* fix(mobile): fix background upload regression on Android by @alextran1502 in https://github.com/immich-app/immich/pull/4139

### Docs
* chore(docs): Fix incorrect header by @bo0tzz in https://github.com/immich-app/immich/pull/4123
* chore(docs): Some SQL queries to copy paste for advanced debugging by @danieldietzler in https://github.com/immich-app/immich/pull/4074
* chore(deps): upgrade sharp and libvips by @mertalev in https://github.com/immich-app/immich/pull/4140

## New Contributors
* @Ploonet made their first contribution in https://github.com/immich-app/immich/pull/4119
* @GenericGuy made their first contribution in https://github.com/immich-app/immich/pull/4128

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.78.1...v1.79.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.79.0'>v1.79.0</a>.</em>

**Comment by mrijke** [comment]
Congrats for shipping the library functionality, looks promising!
Is there any recommended way to migrate from imported assets to a library, without having to add the photos to their corresponding album again?

**Reply by jrasm91** [reply]
There is no official way. The unofficial way is to SQL update the library id of the assets.

**Reply by tspirtle** [reply]
Assuming I'm okay with readding photos to albums/etc after migrating to libraries, is there a recommended way to remove/unlink all the imported assets?
